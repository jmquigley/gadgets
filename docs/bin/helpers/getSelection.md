<a name="getSelectionShim"></a>

## getSelectionShim()
DOM Traversal is not implemented in JSDOM
The best we can do is shim the functions

**Kind**: global function  
