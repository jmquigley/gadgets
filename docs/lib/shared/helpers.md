## Functions

<dl>
<dt><a href="#debug">debug(context, args)</a></dt>
<dd><p>A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.</p>
</dd>
<dt><a href="#globalize">globalize(name, pkg, replace)</a> ⇒ <code>object</code></dt>
<dd><p>Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.</p>
</dd>
</dl>

<a name="debug"></a>

## debug(context, args)
A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.

**Kind**: global function  

| Param | Type | Description |
| --- | --- | --- |
| context | <code>string</code> | the name of the module where this debug request is made. |
| args | <code>Array.&lt;object&gt;</code> | the list of arguments passed to the the debug function. |

<a name="globalize"></a>

## globalize(name, pkg, replace) ⇒ <code>object</code>
Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.

**Kind**: global function  
**Returns**: <code>object</code> - the global instance reference  

| Param | Type | Description |
| --- | --- | --- |
| name | <code>string</code> | the name of the variable to store globally |
| pkg | <code>object</code> | the object associated with the variable name |
| replace | <code>boolean</code> | if true, then it will always overwrite the current global (if it exists) |

