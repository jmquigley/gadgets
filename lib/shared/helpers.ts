"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
