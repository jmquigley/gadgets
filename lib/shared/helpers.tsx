'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
