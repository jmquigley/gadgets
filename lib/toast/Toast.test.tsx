'use strict';

import {cleanup, log, mockupEnv} from '../../test/helpers';
mockupEnv();

import test from 'ava';
import {mount} from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import {Toast, ToastLevel, ToastType, getDefaultToastProps} from './index';

test.after.always.cb(t => {
	cleanup(path.basename(__filename), t);
});

test('Test retrieval of Toast props object', t => {
	const props = getDefaultToastProps();

	t.true('duration' in props);
	t.is(props.duration, 5);

	t.true('level' in props);
	t.is(props.level, ToastLevel.info);

	t.true('type' in props);
	t.is(props.type, ToastType.decay);
});

test('Test creation of a Toast control', t => {
	const ctl = mount(<Toast className="test-class" />);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('id'), '');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.find('.test-class').length, 1);
});
