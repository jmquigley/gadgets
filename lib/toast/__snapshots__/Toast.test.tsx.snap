// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a Toast control 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  contentEditable={false}
  controlled={true}
  decay={true}
  direction="right"
  disabled={false}
  duration={3}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  level={0}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={false}
  notheme={false}
  notooltip={false}
  obj="Toast"
  onClick={[Function]}
  onClose={[Function]}
  onError={[Function]}
  padding={null}
  reset={false}
  selected={false}
  show={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  usebottom={false}
  visible={true}
  width={null}
  xcss={null}
>
  <Toast__ToastView
    className="ui-toast test-class"
    disabled={false}
    level={0}
    style={Object {}}
    usebottom={false}
    visible={false}
    xcss={
      Array [
        "animation:fadeOut ",
        [Function],
        ";opacity:0;z-index:-1;",
      ]
    }
  >
    <Toast__ContentView
      className="ui-toast-content"
      sizing="normal"
    >
      <span />
    </Toast__ContentView>
    <Toast__StyledButton
      iconName="times"
      onClick={[Function]}
      sizing="large"
      style={
        Object {
          "color": "white",
        }
      }
    />
  </Toast__ToastView>
</Wrapper>
`;

exports[`Test retrieval of Toast props object 1`] = `
Object {
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "contentEditable": false,
  "controlled": true,
  "decay": true,
  "direction": "right",
  "disabled": false,
  "duration": 3,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "left": null,
  "level": 0,
  "location": "none",
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Toast",
  "onClick": [Function],
  "onClose": [Function],
  "padding": null,
  "selected": false,
  "show": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "usebottom": false,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;
